product-mini/platforms/zephyr/simple/README.md

```diff
@@ -1,106 +1,82 @@
-
-```shell
-# copy the corresponding board conf file to current directory
-cp boards/qemu_x86_nommu.conf prj.conf
-# then build
-./build_and_run.sh x86
+docker build -t wamr-zephyr .
 ```
 
-> PS: for boards esp32, need to configure some environment first
+Execute the following command to run the image built in the previous step. If
+you are planning to flash a device after building, make sure to specify it with
+[`--device`](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities).
 
 ```shell
-# configure zephyr with espressif
-export ZEPHYR_TOOLCHAIN_VARIANT="espressif"
-export ESPRESSIF_TOOLCHAIN_PATH="/root/.espressif/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/"
-export ESP_IDF_PATH="/root/esp/esp-idf"
-# copy the corresponding board conf file to current directory
-cp boards/esp32.conf prj.conf
-# then build
-./build_and_run.sh esp32
+docker run -it --rm --device=/dev/ttyUSB0 wamr-zephyr
 ```
 
-## Build on local environment
-
-### Dependencies installation
+### Local Environment
 
-Following the Zephyr and Espressif official document:
+Zephyr can also be setup locally to enable building this sample application.
+This allows you have have more control over what modules and tools are
+installed, which can drastically reduce the storage required compared to the
+Docker image.
 
-1. Zephyr installation:
+Follow the steps provided in the [Zephyr Getting Started
+guide](https://docs.zephyrproject.org/latest/develop/getting_started/index.html)
+to setup for local development.
 
-   <https://docs.zephyrproject.org/latest/develop/getting_started/index.html>
+## Building for a Specific Board
 
-2. ESP32 installation:
-
-   <https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/linux-macos-setup.html>
-
-And setup the Zephyr for esp32:
-
-<https://wiki.amarulasolutions.com/zephyr/esp32/esp32-setup.html>
-
-Then Installing QEMU, for example, on Linux:
+With an environment setup either locally or in a Docker container, you can build
+for a Zephyr suppported board using
+[`west`](https://docs.zephyrproject.org/latest/develop/west/index.html). There
+are already [configuaration files](./boards) for a few boards in this sample.
+However, if you are using a new board, you will need to add your own file for
+the board, or define configuration in the [`prj.conf](./prj.conf). After doing
+so, use the following command with your board identifier to build the sample
+application.
 
 ```shell
-sudo apt-get install qemu
+west build . -b <board-identifier> --pristine -- -DWAMR_BUILD_TARGET=<wamr-arch>
 ```
 
-### Run the build script
+The `<board-identifier>` can be found in the Zephyr supported boards
+documentation. It must also be used as the name of the board configuration file.
+You must define the architecture for WAMR to target with `WAMR_BUILD_TARGET`.
+The list of supported architectures can be found in the main project
+[README.md](../../../../README.md#supported-architectures-and-platforms).
```

Comment:
Could you add some examples (e.g. different boards) about how to use the above command?

Reply:
Sure! For context, I would like to follow this PR (and some other updates to the Zephyr support) with CI that tests at least building for a set of target boards that we want to verify. For now, I'll just add some example build commands though 👍🏻